source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -0,0 +1,234 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenOgsXml/OgsXmlGenerator.h>
+#include <MaterialXGenOgsXml/GlslFragmentGenerator.h>
+
+#include <MaterialXGenShader/Shader.h>
+
+#include <MaterialXFormat/XmlIo.h>
+#include <MaterialXFormat/PugiXML/pugixml.hpp>
+
+using namespace pugi;
+
+namespace MaterialX
+{
+
+namespace
+{
+    // Data types used by OGS
+    static const std::unordered_map<const TypeDesc*, const pugi::char_t*> OGS_TYPE_MAP =
+    {
+        { Type::BOOLEAN, "bool" },
+        { Type::FLOAT, "float" },
+        { Type::INTEGER, "int" },
+        { Type::STRING, "int" },
+        { Type::COLOR2, "float2" },
+        { Type::COLOR3, "float3" },
+        { Type::COLOR4, "float4" },
+        { Type::VECTOR2, "float2" },
+        { Type::VECTOR3, "float3" },
+        { Type::VECTOR4, "float4" },
+        { Type::MATRIX33, "float4x4" },
+        { Type::MATRIX44, "float4x4" }
+    };
+
+    // Semantics used by OGS
+    static const std::unordered_map<string, const pugi::char_t*> OGS_SEMANTICS_MAP =
+    {
+        { "positionWorld", "Pw"},
```

Comment:
Just to let you know, it looks like these should be the same as the OGSFX semantics
OGSFX_DEFAULT_SEMANTICS_MAP, so the stream semantics at least I put in incorrectly here.

Reply:
More info. The names of the inputs dictate what space something is in. If it's not using those names then we could either not get any binding to occur or it's just a pass-through (object space).

http://help.autodesk.com/view/MAYAUL/2018/ENU/?guid=__files_GUID_585F5656_4069_4D82_B9BB_3D1AB2D0DFE6_htm

So we'll need to put the "*m" and "*w" words on the list of disallowed names. The documentation isn't correct in that something like Nw is not the semantic but the input name, and the semantic is as listed for OGSFX ("NORMAL" for Nw/Nm).